test/SlowTests/Client/UniqueValues.cs

```diff
@@ -361,5 +368,123 @@ public async Task EnsureCanCompareExchangeWithCorrectCharsEscapeInKey()
             var realNumOfCmpXchg = finalStats.CountOfCompareExchange;
             Assert.Equal(0, realNumOfCmpXchg);
         }
+
+        [Fact]
+        public async Task CanAddMetadataToSimpleCompareExchange()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var str = "Test";
+                var num = 123.456;
+                var key = "egr/test/cmp/x/change/simple";
+                using (var session = store.OpenAsyncSession(new SessionOptions
+                {
+                    TransactionMode = TransactionMode.ClusterWide
+                }))
+                {
+                    var result = session.Advanced.ClusterTransaction.CreateCompareExchangeValue(key, 322);
+                    result.Metadata["TestString"] = str;
+                    result.Metadata["TestNumber"] = num;
+                    await session.SaveChangesAsync();
+                }
+
+                var res = await store.Operations.SendAsync(new GetCompareExchangeValueOperation<int>(key));
+                Assert.NotNull(res.Metadata);
+                Assert.Equal(322, res.Value);
+                Assert.Equal(str, res.Metadata["TestString"]);
+                Assert.Equal(num, res.Metadata["TestNumber"]);
+
+                var stats = await store.Maintenance.SendAsync(new GetDetailedStatisticsOperation());
+                Assert.Equal(1, stats.CountOfCompareExchange);
+            }
+        }
+
+        [Fact]
+        public async Task CanAddMetadataToCompareExchangeAndWaitForExpiration()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var user = new User
+                {
+                    Name = "EGOR"
+                };
+
+                var dateTime = DateTime.Now.AddSeconds(2);
+                var str = "Test";
+                var num = 123.456;
+                var key = "egr/test/cmp/x/change";
+                using (var session = store.OpenAsyncSession(new SessionOptions
+                {
+                    TransactionMode = TransactionMode.ClusterWide
+                }))
+                {
+                    var result = session.Advanced.ClusterTransaction.CreateCompareExchangeValue(key, user);
```

Comment:
What happened if I pass a string as the value here?

Reply:
this should work because its converted to blittable
I added a new test for it.